src/pg/decoder.cr

```diff
@@ -408,23 +408,24 @@ module PG
     struct TimeDecoder
       include Decoder
 
-      DATE_OID = 1082
+      enum OID
+        DATE        = 1082
+        TIMESTAMP   = 1114
+        TIMESTAMPTZ = 1184
+      end
       JAN_1_2K = Time.utc(2000, 1, 1)
 
-      def_oids [
-        DATE_OID, # date
-        1114,     # timestamp
-        1184,     # timestamptz
-      ]
+      def_oids OID.values.map(&.value)
 
       def decode(io, bytesize, oid)
-        if oid == DATE_OID
-          v = read_i32(io)
-          JAN_1_2K + Time::Span.new(days: v, hours: 0, minutes: 0, seconds: 0)
-        else
-          v = read_i64(io) # microseconds
-          sec, m = v.divmod(1_000_000)
-          JAN_1_2K + Time::Span.new(seconds: sec, nanoseconds: m*1000)
+        case oid = OID.new(oid)
+        in .date?
+          JAN_1_2K + read_i32(io).days
+        in .timestamp?
+          JAN_1_2K + read_i64(io).microseconds
+        in .timestamptz?
+          time = JAN_1_2K + read_i64(io).microseconds
+          time.in io.connection.time_zone
```

Comment:
I also noticed we didn't need to `divmod` to split things up for the `Time::Span`. We could just use the microseconds value Postgres sends us and let the `Time::Span` constructor [take care of that work for us](https://github.com/crystal-lang/crystal/blob/29f9ac503fb028e1b095c8f8e036ed74b5474550/src/time/span.cr#L66-L84). It really simplifies this code — for my eyes, at least. 🙂